integration-tests/tests/common/mod.rs

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! This crate contains the common code for integration tests
+
+#![deny(clippy::all)]
+#![deny(missing_docs)]
+
+use std::io::{Error, ErrorKind};
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use std::time::Duration;
+use wait_timeout::ChildExt;
+use walkdir::WalkDir;
+
+pub fn keep_type() -> &'static str {
+    #[cfg(has_sgx)]
+    return "enarx-keep-sgx";
+
+    #[cfg(has_sev)]
+    return "enarx-keep-sev";
+
+    #[cfg(not(any(has_sgx, has_sev)))]
+    compile_error!("Need either SGX or SEV!");
+}
+
+pub struct IntegrationTest {
+    pub wksp_root: PathBuf,
+    pub keep: PathBuf,
+    pub payload: PathBuf,
+    pub shim: PathBuf,
+}
+
+impl IntegrationTest {
+    pub fn new(bin: &str) -> Self {
+        let enarx_dir = Path::new(env!("CARGO_MANIFEST_DIR"));
+        let enarx_dir = enarx_dir.parent().unwrap();
+        let keep_type = keep_type();
+
+        let wksp_root = enarx_dir.to_path_buf();
+        let keep_dir = enarx_dir
+            .join(format!("{}/target/", keep_type))
+            .to_path_buf();
+        let payload_dir = enarx_dir
+            .join("integration-tests/target/x86_64-unknown-linux-musl/")
+            .to_path_buf();
+        let shim_dir = enarx_dir
+            .join(format!(
+                "{}-shim/target/x86_64-unknown-linux-musl/",
+                keep_type
+            ))
+            .to_path_buf();
+        let shim_filename = format!("{}-shim", keep_type);
+        let shim_filename = shim_filename.as_str();
+
+        let keep = find_filepath(keep_dir, keep_type).unwrap();
+        let payload = find_filepath(payload_dir, bin).unwrap();
+        let shim = find_filepath(shim_dir, shim_filename).unwrap();
+
+        Self {
+            wksp_root,
+            keep,
+            payload,
+            shim,
+        }
+    }
+
+    pub fn run(self: Self, timeout: u64, exit_status: i32) -> () {
```

Comment:
This signature is technically correct, but I think it's a little more idiomatic to present this as:

```rust
pub fn run(self, timeout: u64, exit_status: i32) {
```